native_libs/src/parse_csv.c

```diff
@@ -131,30 +131,33 @@ void mat_delete(void **mat)
     free(mat);
 }
 
+void write_csv(const char *filename, char **mat, size_t rows, size_t cols){
+    FILE *fp = fopen(filename, "w+");
+    if(!fp){
+        fprintf(stderr, "%s can't open in %s\n", filename, __func__);
+        perror("fopen");
```

Comment:
You didn't return an error code here, perror does not count, cause it won't be visible anywhere